From feea87f1c2eb33b3cbedb1c07b61006faec8ea32 Mon Sep 17 00:00:00 2001
From: Haithem Salah <haithem.salah@intel.com>
Date: Tue, 5 Apr 2016 14:40:58 +0200
Subject: [PATCH] Remove Silead Power ON/OFF sequence as it introduces touch
 issues on many devices (touch loss after boot) Per L MR1 code line, this
 sequence is not needed, as during resumse the TS driver is getting reset.

Change-Id: I2eb0cc1ee27387e424916ba547688628d93f8e8b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26820
Signed-off-by: Haithem Salah <haithem.salah@intel.com>
Reviewed-on: https://android.intel.com:443/490610
---
 drivers/input/touchscreen/silead.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 2b613b2..e430ea5 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -415,10 +415,6 @@ static int silead_ts_setup(struct i2c_client *client)
 	int ret;
 	u32 status;
 
-	silead_ts_set_power(client, SILEAD_POWER_OFF);
-	msleep(20);
-	silead_ts_set_power(client, SILEAD_POWER_ON);
-	msleep(20);
 
 	ret = silead_ts_get_id(client);
 	if (ret)
@@ -559,9 +555,6 @@ static int silead_ts_resume(struct device *dev)
 	struct i2c_client *client = to_i2c_client(dev);
 	int ret, status;
 
-	/* send power off again, to handle some hardware reset issue */
-	silead_ts_set_power(client, SILEAD_POWER_OFF);
-	silead_ts_set_power(client, SILEAD_POWER_ON);
 
 	ret = silead_ts_reset(client);
 	if (ret) {
-- 
1.9.1

